### ISSUE TITLE
Unable to register a recipe for Steel using IMC

### ISSUE THREAD

**Issue by dshadowwolf** [SlimeKnights/TinkersConstruct#2763]
While helping @jriwanek with some minor issues in [BaseMetals](http://github.com/MinecraftModDevelopment/BaseMetalsl) I started by adding a "liquid coal" so Steel could be made from Iron and Coal in the Smeltery. There was already [an earlier attempt](http://github.com/MinecraftModDevelopment/BaseMetals/src/main/java/com/mcmoddev/basemetals/integration/plugins/TinkersConstruct.java#120) commented out. I fixed up the name of the option and uncommented it. Things worked until I tried running an instance with that version - it broke with an error from Tinkers: 
```slimeknights.tconstruct.library.TinkerAPIException: [TCon API] Could not register Material "steel": It was already registered by Tinkers' Construct```

Basically... Steel is added as a fluid, but trying to add an alloy recipe via the IMC system (in use by @jriwanek to, according to her, avoid a direct dependency on the TiC code) is impossible because the material already exists. I know from looking at the ModTweaker source - and having used ModTweaker and MineTweaker to do it when mods didn't themselves - that it is possible to add an alloy for an existing material.

This seems like a major bug to me - though I don't know if anyone else would see it the same 😃

**Comment by jriwanek** [comment]
Small addition to the info: The IMC was previously for that purpose, I'm intending to move to full integration now I dep on Tcon anyways.

**Comment by bonii-xx** [comment]
Which IMC were you sending? You can't register steel via IMC since it already exists, obviously. See https://github.com/SlimeKnights/TinkersConstruct/wiki/IMC

**Comment by jriwanek** [comment]
https://github.com/MinecraftModDevelopment/BaseMetals/blob/master/src/main/java/com/mcmoddev/basemetals/integration/plugins/TinkersConstruct.java#L122

https://github.com/MinecraftModDevelopment/BaseMetals/blob/master/src/main/java/com/mcmoddev/lib/integration/plugins/TinkersConstruct.java#L69

(Alloy integration, it'd be dumb to try and register the material again)

**Comment by dshadowwolf** [comment]
Steel is registered, but has no recipe. The attempt was using the 'alloy' IMC (as @jriwanek shows in the above links) to register a recipe for it.

**Comment by bonii-xx** [comment]
>(Alloy integration, it'd be dumb to try and register the material again)

Just making sure ;)

Hm, looks weird that a codepath where a material would be registered is called there, it just adds to the general integration management stuff. Will probably require some debugging.

**Comment by bonii-xx** [comment]
Well, I tried to reproduce the issue but it always works. Needs an example NBT or minimal setup to reproduce.

**Comment by dshadowwolf** [comment]
The code is no longer in the tree, but I could pull up the old version and test.

https://github.com/MinecraftModDevelopment/BaseMetals/blob/0fe26947e05525eb3521b1dda92f5cc4c36447e5/src/main/java/com/mcmoddev/basemetals/integration/plugins/TinkersConstruct.java#L122

That's the commented out line that had the call to the function that would register the alloy using the IMC.

https://github.com/MinecraftModDevelopment/BaseMetals/blob/0fe26947e05525eb3521b1dda92f5cc4c36447e5/src/main/java/com/mcmoddev/lib/integration/plugins/TinkersConstruct.java#L69-L93

There is the function that was performing the IMC. As far as I can tell, it is using the format specified in the wiki post you linked to for an "alloy" IMC, As I have tested it against TiC 2.6.1 and 2.6.2 (no numbering on the jar) it might have been fixed in a bug fix released afterwards.
